mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,715 @@
+
+        groups: Optional[ArrayLike] of shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+            By default ``None``.
+
+        z: Optional[ArrayLike] of shape (n_calib_samples, n_exog_features)
+            Exogenous variables
+
+            By default ``None``
+
+        alpha: Optional[float]
+            Between ``0.0`` and ``1.0``, represents the risk level of the
+            confidence interval.
+            Lower ``alpha`` produce larger (more conservative) prediction
+            intervals.
+            ``alpha`` is the complement of the target coverage level.
+
+            If ``None``, the calibration will be done using the ``alpha``value
+            set in the initialisation. Else, the new value will overwrite the
+            old one.
+
+            By default ``None``
+
+        Returns
+        -------
+        MapieCCPRegressor
+            self
+        """
+        self._check_parameters()
+        self._check_calibrate_parameters()
+        check_is_fitted(self, self.fit_attributes)
+
+        if self.cv != 'prefit':
+            self.cv = cast(BaseCrossValidator, self.cv)
+
+            _, calib_index = list(self.cv.split(X, y, groups))[0]
+            (
+                X_calib, y_calib, sample_weight_calib
+            ) = self._safe_sample(X, y, sample_weight, calib_index)
+
+            if z is not None:
+                (
+                    z_calib, _, _
+                ) = self._safe_sample(z, y, sample_weight, calib_index)
+            else:
+                z_calib = None
+        else:
+            X_calib, y_calib, z_calib = X, y, z
+            sample_weight_calib = cast(Optional[NDArray], sample_weight)
+
+        if alpha is not None and self.alpha != alpha:
+            self.alpha = self._check_alpha(alpha)
+            warnings.warn(f"WARNING: The old value of alpha ({self.alpha}) "
+                          f"has been overwritten by the new one ({alpha}).")
+
+        if self.alpha is None:
+            return self
+
+        y_pred_calib = self.estimator_.predict(X_calib)
+
+        calib_conformity_scores = self.conformity_score_.get_conformity_scores(
+            X_calib, y_calib, y_pred_calib
+        )
+
+        if self.conformity_score_.sym:
+            q_low = 1 - self.alpha
+            q_up = 1 - self.alpha
+        else:
+            q_low = self.alpha / 2
+            q_up = 1 - self.alpha / 2
+
+        if self.random_state is None:
+            warnings.warn("WARNING: The method implemented in "
+                          "MapieCCPRegressor has a stochastic behavior. "
+                          "To have reproductible results, use a integer "
+                          "`random_state` value in the `MapieCCPRegressor` "
+                          "initialisation.")
+        else:
+            np.random.seed(self.random_state)
+
+        self.phi_.fit(X, self.estimator_.predict(X), z)
```

Comment:
From 544 to 594, this is the fit full routine of the phi fonction. In order to let the class abstract, you can move it into the phi function class.